src/crystal/system/process.cr

```diff
@@ -0,0 +1,21 @@
+module Crystal
+  # :nodoc:
+  module System
+    # :nodoc:
+    module Process
+      # Returns possible location of executable path
+      # def self.executable_path_impl
+    end
+  end
+end
+
+{% if flag?(:darwin) %}
+  require "./unix/darwin_process"
```

Comment:
It would be better to name the files `process_darwin` etc. to have those process implementations together in lexical sort.